### PR TITLE
[W5.7][T13-3]David Choo

### PR DESCRIPTION
**Pull request by davidchoo12**
Not sure if I'm supposed to split to so many commits, what do you think?

### CODE REVIEW

File: src/seedu/addressbook/data/person/Person.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * Represents a Person in the address book.
  * Guarantees: details are present and not null, field values are validated.
  */
-public class Person implements ReadOnlyPerson {
+public class Person implements ReadOnlyPerson, Comparable {
```
Comment (handshou) [inline src/seedu/addressbook/data/person/Person.java:13]: Consider type-casting Comparable
`public class Person implements ReadOnlyPerson, Comparable<Person> {`

File: src/seedu/addressbook/data/person/Person.java [hunk 2]
```diff
@@ -70,6 +70,16 @@ public void setTags(Set<Tag> replacement) {
         tags.addAll(replacement);
     }
 
+    @Override
+    public int compareTo(Object other) {
```
Comment (handshou) [inline src/seedu/addressbook/data/person/Person.java:74]: After type-casting Comparable, compareTo() takes in Person parameter, rather than Object parameter.

**Review by handshou — COMMENTED** [review]
Great job!

Have some nitpicking on type-casting.

**Comment by Gongjie663746** [comment]
Good job! Thank you so much!

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (NancyQuris) [inline src/seedu/addressbook/parser/Parser.java:14]: I wonder if it is better to specify instead of using just * (not sure, but I remembered it is one of java coding standards). Oh sorry I checked again and realised that it is automatically generated by IntelliJ.

Comment (NancyQuris) [reply src/seedu/addressbook/parser/Parser.java:14]: By the way, Good job!

Comment (damithc) [reply src/seedu/addressbook/parser/Parser.java:14]: Chanced upon this. It's still a coding standard violation. Configure Intellij not to do this.

Comment (NancyQuris) [reply src/seedu/addressbook/parser/Parser.java:14]: Noted, thank you prof!

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -67,6 +67,11 @@ Examples:
 * `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
 * `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
 
+== Sorting all persons : `sort`
+
+Sorts the list of all existing persons by their names. +
+Format: `sort`
```
Comment (parvathy-sudhir-pillai) [inline docs/UserGuide.adoc:73]: Good job updating the user doc

File: src/seedu/addressbook/commands/SortCommand.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sort all persons in the address book.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Persons sorted";
+
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/SortCommand.java:21]: Stray line break? It is a good practice to check the diff of the PR and fix any problems before submitting it.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TestUtil;
+import seedu.addressbook.util.TypicalPersons;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:15]: Coding standard does not allow wild card imports.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 7]
```diff
@@ -43,4 +43,11 @@ private void assertAreUnique(Object... objects) {
     private void assertNotUnique(Object... objects) {
         assertFalse(Utils.elementsAreUnique(Arrays.asList(objects)));
     }
+
+    @Test
+    public void isAnyNull() {
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/common/UtilsTest.java:48]: This commit shouldn't part of this PR. In general, a PR should not contain unmerged code from other PRs. Try to limit each PR to one enhancement, which means the PR should be based on a branch that starts off from the master branch that is in sync with the upstream master branch

**Review by parvathy-sudhir-pillai — APPROVED** [review]
Good effort on this PR @hidingmode. In particular, good job on,
* updating user doc
* updating jUnit tests

Suggestions to improve further:
* update dev doc
* update I/O tests to test the new functionality
* limit the PR to one enhancement and avoid including changes not related to that enhancement or unmerged code from other PRs
* Pay closer attention to the coding standard

Please close the PR after viewing comments.
